config/vault.go

```diff
@@ -233,18 +246,26 @@ func (c *VaultConfig) Finalize() {
 	}
 	c.SSL.Finalize()
 
+	// Order of precedence
+	// 1. `vault_agent_token_file` configuration value
+	// 2. `VAULT_TOKEN` environment variable
+	// 3. `token` configuration value
```

Comment:
I think it is a little odd to go from config file value, to env var, then back to config config file value.

Reply:
Does `token` > `vault_agent_token_file` contents > `VAULT_TOKEN` make more sense? I couldn't find a standard order in the Vault docs for this situation.

Reply:
I think that makes more sense but now am also questioning it.  Maybe explicit tokens should take precedence over the file since that would probably be a more explicit operation.  I don't have strong opinions, so whatever seems to make the most sense.

Reply:
I think maybe `vault_agent_token_file` > `token` > `VAULT_TOKEN` is the way to go. If a `vault_agent_token_file` is being provided it's presumable more up to date than the static `token` field, which should take precedence over the env var.